test/library/packages/BLAS/COMPOPTS

```diff
@@ -0,0 +1 @@
+-L $CRAY_LD_LIBRARY_PATH -g -lgfortran -lsci_gnu
```

Comment:
I think we probably should adjust the link call for LLVM compilations on a PrgEnv system to add this `-L`. See the section of code starting here:

https://github.com/chapel-lang/chapel/blob/master/compiler/llvm/clangUtil.cpp#L3803

which is also supported by `util/config/gather-cray-prgenv-arguments.bash`. If we had another way to get the required libraries etc besides `gather-cray-prgenv-arguments.bash` I think that would be an improvement since that script is pretty hacky.

Reply:
I'm not sure what you're saying -- are you saying to do it in `gather-cray-prgenv-arguments.bash`, or do it in `clangUtil.cpp`?

Or are you saying it requires changes in both?

The original `-L` option was the quickest solution for the BLAS/LAPACK because `$CRAY_LD_LIBRARY_PATH` includes paths to `cray_libsci`'s libraries. Any other libraries which `$CRAY_LD_LIBRARY_PATH` includes are incidental -- if we wanted to be more specific, we could limit it to `-L $CRAY_LIBSCI_PREFIX_DIR/lib`. The only reason I added the `-L` is that it was not finding the `cray-libsci` libraries, which are specific to BLAS/LAPACK.

Reply:
Also I'm not sure `$CRAY_LIBSCI_PREFIX_DIR` will be defined in all configurations, so I chose `$CRAY_LD_LIBRARY_PATH` since it's more generic and always defined on XC systems.

Reply:
I'd recommend adding it to `gather-cray-prgenv-arguments.bash` in the near term but if you realize that this script could be greatly simplified (e.g., in the ideal world, perhaps could be replaced by querying some environment variables, instead of running a trial compile) then it would be great to do so as a follow-up activity.

Reply:
Also it is better to have it here in COMPOPTS here for the very near term - changing gather-cray-prgenv-arguments.bash should be a follow-up PR.

Reply:
https://github.com/Cray/chapel-private/issues/1817